src/editor/parser/marked.js

```diff
@@ -27,7 +27,8 @@ var block = {
   def: /^ *\[([^\]]+)\]: *<?([^\s>]+)>?(?: +["(]([^\n]+)[")])? *(?:\n+|$)/,
   table: noop,
   paragraph: /^((?:[^\n]+\n?(?!hr|heading|lheading|blockquote|tag|def))+)\n*/,
-  text: /^[^\n]+/
+  text: /^[^\n]+/,
+  frontmatter: /^---\n([\s\S]+?)---(?:\n+|$)/
```

Comment:
Maybe it's better to exact match like: `/^---\n[^\n]([\s\S]+?)[^\n]\n------(?:\n+|$)/`.

Reply:
I can not understand why need add `[^\n]`?

Reply:
Because front matter only allows one new line between `---` and the next key-value pair.

```
---
foo: bar
bar: foo
---
```

Reply:
Can you give me a `bad` example that front matter not allowed?

Reply:
You allow to much. I don't know if it could cause problems with markdown. You RegEx allows for example:

```
---
a---
```

or

```
---

text



----
```

Reply:
just to see weather user will encounter the same problem, then we'll fixed it.